app/actions/app_apply_manifest.rb

```diff
@@ -193,17 +191,21 @@ def create_binding_message(service_instance_guid, app_guid, manifest_service_bin
       )
     end
 
-    def raise_invalid_binding_error!(service_instance, binding)
-      case binding.last_operation.state
-      when 'in progress'
-        error_message = "For service '#{service_instance.name}': A binding is being #{binding.last_operation.type}d. Retry this operation later."
-      when 'failed'
-        error_message = "For service '#{service_instance.name}': A binding failed to be #{binding.last_operation.type}d. "\
-                        'Resolve the issue with this binding before retrying this operation.'
-      end
+    def raise_binding_error!(service_instance, message)
+      error_message = "For service '#{service_instance.name}': #{message}"
       raise ServiceBindingError.new(error_message)
     end
 
+    def raise_binding_operation_in_progress!(service_instance, operation)
+      error_message = "A binding is being #{operation}d. Retry this operation later."
+      raise_binding_error!(service_instance, error_message)
+    end
+
+    def raise_binding_deletion_failed!(service_instance, operation)
+      error_message = "A binding failed to be #{operation}d. Resolve the issue with this binding before retrying this operation."
```

Comment:
One very small thing: the method is called `raise_binding_deletion_failed!`, so it's purpose is to raise an exception when the state is "delete failed". Thus the `operation` parameter is not required; it's always `delete`.